#!/usr/bin/env python
PACKAGE = "cgal_raytracer"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("minAngleH",   double_t,   1, "Minimum horizontal angle.", -45, -90, 90)
gen.add("maxAngleH",   double_t,   2, "Maximum horizontal angle.",  45, -90, 90)
gen.add("minAngleV",   double_t,   4, "Minimum vertical angle.",   -30, -90, 90)
gen.add("maxAngleV",   double_t,   8, "Maximum vertical angle.",    30, -90, 90)
gen.add("resolutionH", int_t,     16, "Horizontal resolution",     135,   1, 200)
gen.add("resolutionV", int_t,     32, "Vertical resolution",        90,   1, 200)
# this doesn't affect the camera much
gen.add("focalLength", double_t,  64, "Focal length",                0.4, 0, 5)
# the ranges are not used at all
gen.add("minRange",    double_t, 128, "Minimum scan range",          0,   0, 10)
gen.add("maxRange",    double_t, 256, "Maximum scan range",          8,   0, 10)
gen.add("rt_frame",    str_t,    512, "Frame for the PointCloud.",   "rt_frame")

exit(gen.generate(PACKAGE, "cgal_raytracer", "CamParam"))
